src/Features/Core/Portable/CodeRefactorings/AddMissingImports/AbstractAddMissingImportsRefactoringProvider.cs

```diff
@@ -3,33 +3,38 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
+using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.CodeActions;
 using Microsoft.CodeAnalysis.CodeCleanup;
 using Microsoft.CodeAnalysis.CodeRefactorings;
 using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Host.Mef;
 using Microsoft.CodeAnalysis.PasteTracking;
 using Microsoft.CodeAnalysis.Shared.Extensions;
 
 namespace Microsoft.CodeAnalysis.AddMissingImports
 {
-    internal abstract class AbstractAddMissingImportsRefactoringProvider(IPasteTrackingService? pasteTrackingService) : CodeRefactoringProvider
+    internal abstract class AbstractAddMissingImportsRefactoringProvider : CodeRefactoringProvider
     {
-        private readonly IPasteTrackingService? _pasteTrackingService = pasteTrackingService;
         protected abstract string CodeActionTitle { get; }
 
         public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
         {
-            // If we aren't in a host that supports paste tracking, we can't do anything. This is just to avoid creating MEF part rejections for
+            var (document, _, cancellationToken) = context;
+
+            // If we aren't in a host that supports paste tracking (known by having exactly one export of type
+            // IPasteTrackingService), we can't do anything. This is just to avoid creating MEF part rejections for
             // things composing the Features layer.
-            if (_pasteTrackingService == null)
+            var services = document.Project.Solution.Workspace.Services.HostServices as IMefHostExportProvider;
+            var pasteTrackingService = services?.GetExports<IPasteTrackingService>().SingleOrDefault()?.Value;
```

Comment:
always wary about SingleOrDefault (throws if there are multiple).  can we tweak that, or is it desirale to throw in that case?

Reply:
It seems desirable to throw in that case rather than silently have the feature stop working.